Modules/Devices/test/Devices.Application.Tests/Tests/Devices/Commands/DeleteDevice/HandlerTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.UserContext;
+using Backbone.Modules.Devices.Application.Devices.Commands.DeleteDevice;
+using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Devices.Domain.Entities;
+using FakeItEasy;
+using FluentAssertions;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace Backbone.Modules.Devices.Application.Tests.Tests.Devices.Commands.DeleteDevice;
+public class HandlerTests
+{
+    [Fact]
+    public async Task Deletes_device_owned_by_identity()
+    {
+        // Arrange
+        var startTime = DateTime.UtcNow;
+
+        var identity = TestDataGenerator.CreateIdentity();
+        var device = new Device(identity);
+
+        var mockIdentitiesRepository = A.Fake<IIdentitiesRepository>();
+        A.CallTo(() => mockIdentitiesRepository.GetDeviceById(device.Id, A<CancellationToken>._, A<bool>._)).Returns(device);
+        A.CallTo(() => mockIdentitiesRepository.Update(device, A<CancellationToken>._)).DoesNothing();
+
+        var fakeUserContext = A.Fake<IUserContext>();
+        A.CallTo(() => fakeUserContext.GetAddress()).Returns(identity.Address);
+        A.CallTo(() => fakeUserContext.GetDeviceId()).Returns(device.Id);
+
+        var deleteDeviceCommand = new DeleteDeviceCommand
+        {
+            DeviceId = device.Id
+        };
+
+        var dummyLogger = A.Fake<ILogger<Handler>>();
+
+        var handler = new Handler(mockIdentitiesRepository, fakeUserContext, dummyLogger);
+
+        // Act
+        await handler.Handle(deleteDeviceCommand, CancellationToken.None);
+
+        // Assert
+        A.CallTo(() => mockIdentitiesRepository.GetDeviceById(
+            device.Id,
+            A<CancellationToken>._,
+            A<bool>._
+        )).MustHaveHappenedOnceExactly();
```

Comment:
No need to assert this. It's overspecification (https://dev.to/edgaremmanuel/writing-maintainable-tests-overspecification-in-unit-tests-2lak). We don't care how the handler gets the device.

Reply:
Alright, I was thinking along the lines of checking if the test "ticked all the boxes" in the `Handle(...)` method.

Reply:
Yeah, but that's more or less the definition of overspecification. ;) You should read the linked site (if you didn't already do it)